### PR TITLE
docs: rfc for evaluation for updated features

### PR DESCRIPTION
**Pull request by masaaania**

### CODE REVIEW

File: docs/rfcs/0042-evaluation-for-updated-features.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+As an exception, the following feature flags must be evaluated regardless of the timestamp value.
+- Feature flags that depend on the feature flags that need to be evaluated
+- (if user attributes have been updated) Feature flags with targeting rules
+
+If user attributes have been updated, the result of the evaluation may also change.
+Since the server has to know that update, the SDK sends the `IsUserAttributesChanged` flag to the server as a request parameter like the timestamp.
+
+In addition, the following changes are required in both the server and SDK implementations:
+- Since only the updated feature flags are evaluated and returned, the SDK's implementation for updating local data needs to be changed.
+- The server needs to return information about archived feature flags to the SDK.
```
Comment (ono-max) [inline docs/rfcs/0042-evaluation-for-updated-features.md:31]: Sorry, I can not understand why the server need to return archived feature flags. Could you write the reason?

File: docs/rfcs/0042-evaluation-for-updated-features.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  - So the server must put the archived feature flags to Redis.
+- The validation of the request parameter `Tag` needs to be removed.
+
+
+### Evaluation on Server
```
Comment (ono-max) [inline docs/rfcs/0042-evaluation-for-updated-features.md:36]: Can you write a description of how to evaluate feature flags currently for feature contributors?

**Comment by ono-max** [comment]
@masaaania 

Can you write the change in protobuf?

File: docs/rfcs/0042-evaluation-for-updated-features.md (hunk 1 above)
Comment (masaaania) [reply docs/rfcs/0042-evaluation-for-updated-features.md]: Thank you for your comment, I added it 👍

File: docs/rfcs/0042-evaluation-for-updated-features.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+When GetEvaluations is called again, that timestamp is sent to the server as a request parameter.
+
+The server checks it against the updatedAt value of the feature flags, and only evaluates those that have been updated since the previous evaluation.
+
+As an exception, the following feature flags must be evaluated regardless of the timestamp value.
+- Feature flags that depend on the feature flags that need to be evaluated
+- (if user attributes have been updated) Feature flags with targeting rules
+
+If user attributes have been updated, the result of the evaluation may also change.
+Since the server has to know that update, the SDK sends the `IsUserAttributesChanged` flag to the server as a request parameter like the timestamp.
```
Comment (ono-max) [inline docs/rfcs/0042-evaluation-for-updated-features.md:27]: I think that UserAttributesUpdated is better. What do you think?

Comment (masaaania) [reply docs/rfcs/0042-evaluation-for-updated-features.md:27]: I think so too 👍 

SDK has the function named `updateUserAttributes`.
https://github.com/bucketeer-io/android-client-sdk/blob/73508d1642da3c6db052f66a7471032aad56c269/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/BKTClient.kt#L28

**Review by ono-max — DISMISSED** [review]
👍

**Review by ono-max — DISMISSED** [review]

**Review by cre8ivejp — APPROVED** [review]
Thank you for this 🎉

**Pull request merged**